### ISSUE TITLE
Doesn't work

### ISSUE THREAD

**Issue by cyberfox1** [functionalone/serverless-iam-roles-per-function#51]
All lambdas get a default role associated with stack rather than the custom one per lambda. Copied example code.

Using serverless 2.0.0.

**Comment by VMois** [comment]
Please, provide code here + version of the library.

**Comment by jfoxtech** [comment]
Allow me to provide an example:

Code snippet:

```
  events-buffered-topic-posts:
    handler: bin/events-buffered-topic-posts
    name: events-buffered-topic-posts-${opt:stage}
    description: "Call topic-post service when assets are done processing"
    package:
      exclude:
        - ./**
      include:
        - ./bin/events-buffered-topic-posts
    iamRoleStatements:
      - Effect: Allow
        Action:
          - "lambda:InvokeFunction"
        Resource: [
          "arn:aws:lambda:us-east-1:655063978684:function:topic-post-service-${opt:stage}",
          "arn:aws:lambda:us-east-1:655063978684:function:buffer-service-${opt:stage}"
        ]
```

Output:

```
cd src/places/; serverless deploy --stage "jfox" --infraenv "dev" -v --force
Serverless: Running "serverless" installed locally (in service node_modules)
Serverless: Configuration warning:
Serverless:   at 'functions['place-service']': unrecognized property 'iamRoleStatements'
Serverless:   at 'functions['place-stream']': unrecognized property 'iamRoleStatements'
Serverless:   at 'functions['place-data']': unrecognized property 'iamRoleStatements'
Serverless:  
Serverless: Learn more about configuration validation here: http://slss.io/configuration-validation
Serverless:  
Serverless: Packaging service...
Serverless: Excluding development dependencies...
Serverless: Excluding development dependencies...
Serverless: Excluding development dependencies...
```

Any help on this would be much appreciated.

**Comment by enapupe** [comment]
I'm having the same issue: 
```
Serverless Error ---------------------------------------

  Configuration error at 'functions.twitchUpdateBearer': unrecognized property 'iamRoleStatements'
```

using
├─ serverless@2.9.0
└─ serverless-iam-roles-per-function@2.0.2

**Comment by enapupe** [comment]
Actually, not a blocker, see https://github.com/serverless/serverless/issues/8422#issuecomment-716184282

**Comment by bodhiprice** [comment]
I ran into this issue as well and I thought I'd share a bit more info for anyone else who has had their deploys stop working with this error - `unrecognized property 'iamRoleStatements'`.

It looks like the serverless team released a breaking change on a minor release, which is pretty frustrating. The quote: `Function definition schema validation has been enforced from v2.7.0.`. So a lot of plugins broke with that 2.7 release. @enapupe has linked to the issue where it's discussed above, but I don't understand the comment that it's not a blocker. I had to go back to a previous version of serverless to get it going, which I think is the solution until the various plugins this affects are updated.

Full issue here: https://github.com/serverless/serverless/issues/8422

Really seems like this should have been part of a major release if I'm understanding things correctly.
